hw/top_earlgrey/syn/constraints.sdc

```diff
@@ -64,6 +68,45 @@ set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dp_en_o]        -clock ${USB_C
 set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dn_o]           -clock ${USB_CLK_PIN}
 set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dn_en_o]        -clock ${USB_CLK_PIN}
 
+#####################
+# IO clk (24MHz)    #
+#####################
+set IO_CLK_PIN clk_io_i
+set IO_TCK 40
+set_ideal_network ${IO_CLK_PIN}
+
+create_clock ${IO_CLK_PIN} -period ${IO_TCK}
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} ${IO_CLK_PIN}
+
+set IN_DEL  IO_TCK/2
+set OUT_DEL IO_TCK/2
+
+set_input_delay ${IN_DEL} [get_ports mio_in_i*]          -clock ${IO_CLK_PIN}
+
+set_output_delay ${OUT_DEL} [get_ports mio_out_o*]       -clock ${IO_CLK_PIN}
+set_output_delay ${OUT_DEL} [get_ports mio_oe_o*]        -clock ${IO_CLK_PIN}
+
+# UART RX
```

Comment:
Are the constants here (10, 9 etc) supposed to reference the variables defined above? such as `PORT_UART_RX`?
This also reminds me...do you think it makes sense as part of `topgen`, we actually generate a tcl command set of the variables above?

Reply:
Oh.. just revised. 

I totally agree that some portion of this constraint can be generated by topgen, removing human error.

Reply:
Indeed there is still quite some potential for topgen optimization. 

On a side note, I also do not like that we have multiple tops lying around which are all largely the same and hand edited. This just leads to bugs which are hard to find (we've just seen another instance of this)...